Flaky test in nyc's test suite

<!--Please use the template provided below when reporting bugs:-->
## Link to bug demonstration repository
<!--
A demo repository will show the combination of module versions being used,
command-lines and configurations.  NYC is very complex, it is difficult or
even impossible to triage most bugs without this information.

If you cannot share a link to your repository (maybe it's private) then you must
create a basic `nyc-bug-demo` repository and link to that please [mandatory].
-->
This one ;)

I'm finding one of the tests is flaky in the nyc test suite.  The test is `'nyc displays help to stderr when it doesn\'t know what to do'` in `test/nyc-integration.js`.  It fails on the check `t.equal(stderr, help.stdout)`.

## Expected Behavior
Running `npm t` on nyc's current master repo produces consistent results
## Observed Behavior
Running `npm t` on nyc's current master repo produces inconsistent results due to a flaky test,
`'nyc displays help to stderr when it doesn\'t know what to do'`, in `test/nyc-integration.js`.

### Troubleshooting steps

## Environment Information
<!--
[mandatory] run the following script: 
  npx envinfo@latest --preset nyc
-->
```
npx: installed 1 in 0.814s

  System:
    OS: macOS High Sierra 10.13.6
    CPU: (4) x64 Intel(R) Core(TM) i5-7600 CPU @ 3.50GHz
    Memory: 7.60 GB / 32.00 GB
  Binaries:
    Node: 10.17.0 - ~/.nvm/versions/node/v10.17.0/bin/node
    Yarn: 1.21.1 - /usr/local/bin/yarn
    npm: 6.13.4 - ~/.nvm/versions/node/v10.17.0/bin/npm
  npmPackages:
    istanbul-lib-coverage: ^3.0.0 => 3.0.0 
    istanbul-lib-hook: ^3.0.0 => 3.0.0 
    istanbul-lib-instrument: ^4.0.0 => 4.0.0 
    istanbul-lib-processinfo: ^2.0.2 => 2.0.2 
    istanbul-lib-report: ^3.0.0 => 3.0.0 
    istanbul-lib-source-maps: ^4.0.0 => 4.0.0 
    istanbul-reports: ^3.0.0 => 3.0.0 
    source-map-support: ^0.5.16 => 0.5.16 
```

Example of a passing and failing test...
```shell
iMac-3:nyc andrewf$ npm t

> nyc@15.0.0 pretest /Users/andrewf/work/Github_projects/nyc
> npm run lint && npm run clean && npm run instrument


> nyc@15.0.0 lint /Users/andrewf/work/Github_projects/nyc
> standard


> nyc@15.0.0 clean /Users/andrewf/work/Github_projects/nyc
> node ./npm-run-clean.js


> nyc@15.0.0 instrument /Users/andrewf/work/Github_projects/nyc
> node ./build-self-coverage.js


> nyc@15.0.0 test /Users/andrewf/work/Github_projects/nyc
> tap

 PASS  test/add-all-files.js 13 OK 2s
 PASS  test/cache.js 2 OK 4s
 PASS  test/config-override.js 3 OK 1s
 PASS  test/config.js 15 OK 84.54ms
 PASS  test/cwd.js 4 OK 63.739ms
 PASS  test/eager.js 6 OK 1s
 PASS  test/instrument.js 64 OK 9s
 PASS  test/issue-190.js 4 OK 4s
 FAIL  test/nyc-integration.js
 ✖ should be equal

  test/nyc-integration.js                                             
  295 |   t.equal(status, 1)                                          
  296 |   t.equal(stdout, '')                                         
> 297 |   t.equal(stderr, help.stdout)                                
      | ----^                                                         
  298 | })                                                            
  299 |                                                               
  300 | t.test('handles --clean / --no-clean properly', async t => {  

  --- expected                                                                     
  +++ actual                                                                       
  @@ -111,5 +111,8 @@                                                              
     nyc.js npm test                           instrument your tests with coverage 
     nyc.js --require @babel/register npm      instrument your tests with coverage 
     test                                      and transpile with Babel            
     nyc.js report --reporter=text-lcov        output lcov report after running    
  -                                                                                
  +                                            your tests                          
  +                                                                                
  +visit https://git.io/vHysA for list of available reporters                      
  +                                                                                

  test: test/nyc-integration.js nyc displays help to stderr when it doesn't know what
    to do
  stack: |
    Test.t.test (test/nyc-integration.js:297:5)

 FAIL  test/nyc-integration.js 1 failed of 221 14s
 ✖ should be equal

 PASS  test/parser-plugins.js 6 OK 835.683ms
 PASS  test/process-args.js 5 OK 26.982ms
 PASS  test/processinfo.js 21 OK 2s
 PASS  test/report.js 3 OK 1s
 PASS  test/should-instrument.js 23 OK 98.925ms
 PASS  test/source-map-support.js 2 OK 441.032ms
 PASS  test/temp-dir.js 15 OK 1s
 PASS  test/tsc.js 6 OK 1s
 PASS  test/wrap.js 12 OK 577.593ms

                         
  🌈 SUMMARY RESULTS 🌈  
                         

 FAIL  test/nyc-integration.js 1 failed of 221 14s
 ✖ should be equal

Suites:   1 failed, 17 passed, 18 of 18 completed
Asserts:  1 failed, 424 passed, of 425
Time:     47s
npm ERR! Test failed.  See above for more details.
iMac-3:nyc andrewf$ npm t

> nyc@15.0.0 pretest /Users/andrewf/work/Github_projects/nyc
> npm run lint && npm run clean && npm run instrument


> nyc@15.0.0 lint /Users/andrewf/work/Github_projects/nyc
> standard


> nyc@15.0.0 clean /Users/andrewf/work/Github_projects/nyc
> node ./npm-run-clean.js


> nyc@15.0.0 instrument /Users/andrewf/work/Github_projects/nyc
> node ./build-self-coverage.js


> nyc@15.0.0 test /Users/andrewf/work/Github_projects/nyc
> tap

 PASS  test/add-all-files.js 13 OK 3s
 PASS  test/cache.js 2 OK 5s
 PASS  test/config-override.js 3 OK 1s
 PASS  test/config.js 15 OK 86.122ms
 PASS  test/cwd.js 4 OK 59.56ms
 PASS  test/eager.js 6 OK 1s
 PASS  test/instrument.js 64 OK 11s
 PASS  test/issue-190.js 4 OK 4s
 PASS  test/nyc-integration.js 221 OK 14s
 PASS  test/parser-plugins.js 6 OK 787.448ms
 PASS  test/process-args.js 5 OK 32.526ms
 PASS  test/processinfo.js 21 OK 2s
 PASS  test/report.js 3 OK 1s
 PASS  test/should-instrument.js 23 OK 111.272ms
 PASS  test/source-map-support.js 2 OK 417.408ms
 PASS  test/temp-dir.js 15 OK 1s
 PASS  test/tsc.js 6 OK 1s
 PASS  test/wrap.js 12 OK 798.066ms

                         
  🌈 SUMMARY RESULTS 🌈  
                         

Suites:   18 passed, 18 of 18 completed
Asserts:  425 passed, of 425
Time:     51s

> nyc@15.0.0 posttest /Users/andrewf/work/Github_projects/nyc
> npm run report


> nyc@15.0.0 report /Users/andrewf/work/Github_projects/nyc
> node ./bin/nyc report --temp-dir ./.self_coverage/ -r text -r lcov

-----------------------|---------|----------|---------|---------|-------------------
File                   | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
-----------------------|---------|----------|---------|---------|-------------------
All files              |     100 |      100 |     100 |     100 |                   
 nyc                   |     100 |      100 |     100 |     100 |                   
  index.js             |     100 |      100 |     100 |     100 |                   
 nyc/bin               |     100 |      100 |     100 |     100 |                   
  nyc.js               |     100 |      100 |     100 |     100 |                   
  wrap.js              |     100 |      100 |     100 |     100 |                   
 nyc/lib               |     100 |      100 |     100 |     100 |                   
  config-util.js       |     100 |      100 |     100 |     100 |                   
  fs-promises.js       |     100 |      100 |     100 |     100 |                   
  hash.js              |     100 |      100 |     100 |     100 |                   
  process-args.js      |     100 |      100 |     100 |     100 |                   
  register-env.js      |     100 |      100 |     100 |     100 |                   
  source-maps.js       |     100 |      100 |     100 |     100 |                   
  wrap.js              |     100 |      100 |     100 |     100 |                   
 nyc/lib/commands      |     100 |      100 |     100 |     100 |                   
  check-coverage.js    |     100 |      100 |     100 |     100 |                   
  helpers.js           |     100 |      100 |     100 |     100 |                   
  instrument.js        |     100 |      100 |     100 |     100 |                   
  merge.js             |     100 |      100 |     100 |     100 |                   
  report.js            |     100 |      100 |     100 |     100 |                   
 nyc/lib/instrumenters |     100 |      100 |     100 |     100 |                   
  istanbul.js          |     100 |      100 |     100 |     100 |                   
  noop.js              |     100 |      100 |     100 |     100 |                   
-----------------------|---------|----------|---------|---------|-------------------
```

--- a/test/nyc-integration.js
+++ b/test/nyc-integration.js
@@ -294,7 +294,7 @@ t.test('nyc displays help to stderr when it doesn\'t know what to do', async t =
 
   t.equal(status, 1)
   t.equal(stdout, '')
-  t.equal(stderr, help.stdout)
+  t.match(stderr, help.stdout)
 })
 
 t.test('handles --clean / --no-clean properly', async t => {

